src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -40,6 +40,28 @@ public override NoPia.EmbeddedTypesManager EmbeddedTypesManagerOpt
         /// </summary>
         private Dictionary<FieldSymbol, NamedTypeSymbol> _fixedImplementationTypes;
 
+        private bool _needsGeneratedIsReadOnlyAttribute_IsFrozen;
+        private bool _needsGeneratedIsReadOnlyAttribute_Value;
+
+        /// <summary>
+        /// Returns a value indicating whether this builder has a symbol that needs IsReadOnlyAttribute to be generated during emit phase.
+        /// The value is set during during lowering the symbols that need that attribute, and is frozen on first trial to get it.
+        /// Freezing is needed to make sure that nothing tries to modify the value after the value is read.
+        /// </summary>
+        internal bool NeedsGeneratedIsReadOnlyAttribute
```

Comment:
Yes. Because it is set by the module builder when we're building a module as well. that's the object we pass around during emit phase. Afterwards, this produces errors if it needs generating, and the child class actually generates it.

---
In reply to: [115032919](https://github.com/dotnet/roslyn/pull/18715#discussion_r115032919) [](ancestors = 115032919)